Product/Production/Services/DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/docsubmission/audit/transform/DSDeferredResponseAuditTransforms.java

```diff
@@ -0,0 +1,231 @@
+
+    @Override
+    protected String getServiceEventDisplayRequestor() {
+        return DocSubmissionAuditTransformsConstants.EVENT_ID_DISPLAY_RECIPIENT;
+    }
+
+    @Override
+    protected String getServiceEventDisplayResponder() {
+        return DocSubmissionAuditTransformsConstants.EVENT_ID_DISPLAY_SOURCE;
+    }
+
+    @Override
+    protected String getServiceEventTypeCode() {
+        return DocSubmissionAuditTransformsConstants.EVENT_TYPE_CODE;
+    }
+
+    @Override
+    protected String getServiceEventTypeCodeSystem() {
+        return DocSubmissionAuditTransformsConstants.EVENT_TYPE_CODE_SYSTEM;
+    }
+
+    @Override
+    protected String getServiceEventTypeCodeDisplayName() {
+        return DocSubmissionAuditTransformsConstants.EVENT_TYPE_CODE_DISPLAY_NAME;
+    }
+
+    @Override
+    protected String getServiceEventActionCodeRequestor() {
+        return DocSubmissionAuditTransformsConstants.EVENT_ACTION_CODE_RECIPIENT;
+    }
+
+    @Override
+    protected String getServiceEventActionCodeResponder() {
+        return DocSubmissionAuditTransformsConstants.EVENT_ACTION_CODE_SOURCE;
+    }
+
+    @Override
+    protected AuditMessageType.ActiveParticipant getActiveParticipant(UserType oUserInfo) {
+        return null;
+    }
+
+    @Override
+    protected AuditMessageType.ActiveParticipant getActiveParticipantSource(NhinTargetSystemType target,
+        String serviceName, boolean isRequesting, Properties webContextProperties) {
+
+        String ipOrHost = isRequesting ? getDSDeferredRequestInitiatorAddress(target) : getLocalHostAddress();
+
+        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+        participant.setUserID(isRequesting ? NhincConstants.WSA_REPLY_TO
+            : getInboundReplyToFromHeader(webContextProperties));
+        if (isRequesting) {
+            participant.setAlternativeUserID(ManagementFactory.getRuntimeMXBean().getName());
+        }
+        participant.setNetworkAccessPointID(ipOrHost);
+        participant.setNetworkAccessPointTypeCode(getNetworkAccessPointTypeCode(ipOrHost));
+        participant.getRoleIDCode().add(AuditDataTransformHelper.createCodeValueType(
+            AuditTransformsConstants.ACTIVE_PARTICIPANT_ROLE_CODE_SOURCE, null,
+            AuditTransformsConstants.ACTIVE_PARTICIPANT_CODE_SYSTEM_NAME,
+            AuditTransformsConstants.ACTIVE_PARTICIPANT_ROLE_CODE_SOURCE_DISPLAY_NAME));
+        participant.setUserIsRequestor(Boolean.TRUE);
+        return participant;
+
+    }
+
+    @Override
+    protected AuditMessageType.ActiveParticipant getActiveParticipantDestination(NhinTargetSystemType target,
+        boolean isRequesting, Properties webContextProperties, String serviceName) {
+
+        String url;
+        String ipOrHost;
+
+        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+
+        url = isRequesting ? getWebServiceUrlFromRemoteObject(getLocalHCIDNhinTarget(),
```

Comment:
Unless there's a reason for them to be split, can you combine declaration and assignment for url?